src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -0,0 +1,236 @@
+        commandParams.hardwareVersion = [NSNumber numberWithInt:commandData.hardwareVersion.Value()];
+    }
+
+    if (commandData.location.HasValue()) {
+        commandParams.location = [NSString stringWithUTF8String:commandData.location.Value().data()];
+    }
+
+    if (commandData.requestorCanConsent.HasValue()) {
+        commandParams.requestorCanConsent = [NSNumber numberWithBool:commandData.requestorCanConsent.Value()];
+    }
+
+    if (commandData.metadataForProvider.HasValue()) {
+        commandParams.metadataForProvider = [NSData dataWithBytes:commandData.metadataForProvider.Value().data()
+                                                           length:commandData.metadataForProvider.Value().size()];
+    }
+}
+
+void MTROTAProviderDelegateBridge::ConvertFromQueryImageResponseParms(
+    const MTROtaSoftwareUpdateProviderClusterQueryImageResponseParams * responseParams,
+    chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::QueryImageResponse::Type & response)
+{
+    response.status = static_cast<chip::app::Clusters::OtaSoftwareUpdateProvider::OTAQueryStatus>([responseParams.status intValue]);
+
+    if (responseParams.delayedActionTime) {
+        response.delayedActionTime.SetValue([responseParams.delayedActionTime intValue]);
+    }
+
+    if (responseParams.imageURI) {
+        response.imageURI.SetValue(chip::CharSpan([responseParams.imageURI UTF8String], responseParams.imageURI.length));
+    }
+
+    if (responseParams.softwareVersion) {
+        response.softwareVersion.SetValue([responseParams.softwareVersion intValue]);
+    }
+
+    if (responseParams.softwareVersionString) {
+        response.softwareVersionString.SetValue(
+            chip::CharSpan([responseParams.softwareVersionString UTF8String], responseParams.softwareVersionString.length));
+    }
+
+    if (responseParams.updateToken) {
+        UInt8 * updateTokenBytes = (UInt8 *) responseParams.updateToken.bytes;
+        response.updateToken.SetValue(chip::ByteSpan(updateTokenBytes, responseParams.updateToken.length));
+    }
+
+    if (responseParams.userConsentNeeded) {
+        response.userConsentNeeded.SetValue([responseParams.userConsentNeeded boolValue]);
+    }
+
+    if (responseParams.metadataForRequestor) {
+        UInt8 * metadataForRequestorBytes = (UInt8 *) responseParams.metadataForRequestor.bytes;
+        response.metadataForRequestor.SetValue(
+            chip::ByteSpan(metadataForRequestorBytes, responseParams.metadataForRequestor.length));
+    }
+}
+
+void MTROTAProviderDelegateBridge::ConvertToApplyUpdateRequestParams(
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::ApplyUpdateRequest::DecodableType & commandData,
+    MTROtaSoftwareUpdateProviderClusterApplyUpdateRequestParams * commandParams)
+{
+    commandParams.updateToken = [NSData dataWithBytes:commandData.updateToken.data() length:commandData.updateToken.size()];
+    commandParams.newVersion = [NSNumber numberWithInt:commandData.newVersion];
```

Comment:
`numberWithUnsignedLong`.  I wish this code were compiled with `-Wconversion`.  :(